packages/flutter/test/material/dropdown_test.dart

```diff
@@ -1080,4 +1082,18 @@ void main() {
     await tester.pumpAndSettle();
     expect(selectedIndex, 13);
   });
+
+  testWidgets('Dropdown custom underline golden', (WidgetTester tester) async {
```

Comment:
Rather than using a golden image test here, it would sufficient to check the decoration of the DecoratedBox widget below the DropDownButton. Golden image tests failures can be difficult to diagnose and they're expensive to maintain.

Something like:
```
Finder decoratedBox = find.descendant(
  of: find.byType(DropdownButton),
  matching: find.byType(DecoratedBox),
);
expect(tester.widget(decoratedBox).decoration, ...)
```

Reply:
Agree with @HansMuller  here, this'll make the tests more reliable in the future if any changes have to be made to the widget's implementation. You could reference how I approached similar tests I made with Icon colors in #29572

Reply:
Sounds good. I've removed the golden test and added a test checking the DecoratedBox of the DropdownButton.

Also, thanks for the feedback for a newcomer @HansMuller and @shihaohong